openlibrary/plugins/upstream/account.py

```diff
@@ -1134,14 +1134,27 @@ class my_follows(delegate.page):
     path = r"/people/([^/]+)/(followers|following)"
 
     def GET(self, username, key=""):
-        mb = MyBooksTemplate(username, 'following')
+        i = web.input(page=1)
+        page_size = 25
+        page = 1 if not i.page.isdigit() else max(1, int(i.page))
+        offset = (page - 1) * page_size
```

Comment:
Since the page is also used to calculate the offset, I validate it here and pass it into the template instead of defining in the template. The number of values being passed to the template is quite large, and this might not be totally necessary, but doing it this way lets me force invalid page values in the query param to default to page 1 (i.e. ?page=0, ?page=-1, ?page=aaaa all default to the first page) before using it to calculate the offset.